### DISCUSSION TITLE
Issue with Dexing and Deprecated API in record Package on Android Build

### DISCUSSION THREAD

**Discussion by Nikunj-Flutter** [llfbandit/record#434] [Q&A]
Hi,

I am encountering an issue while building my Flutter app with the record package. Specifically, I am getting the following error related to Dexing and a deprecated API when trying to build the Android app:

`Note: C:\Users\nikunj\AppData\Local\Pub\Cache\hosted\pub.dev\contacts_service-0.6.3\android\src\main\java\flutter\plugins\contactsservice\contactsservice\ContactsServicePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
ERROR:C:\Users\nikunj\Desktop\work\help_app\build\record_android\intermediates\runtime_library_classes_jar\debug\classes.jar: D8: com.android.tools.r8.internal.Hc: Sealed classes are not supported as program classes
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeLibDexDebug'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform classes.jar (project :record_android) to match attributes {artifactType=android-dex, asm-transformed-variant=NONE, com.android.build.api.attributes.AgpVersionAttr=7.3.0, com.android.build.api.attributes.BuildTypeAttr=debug, com.android.build.gradle.internal.attributes.VariantAttr=debug, dexing-enable-desugaring=true, dexing-enable-jacoco-instrumentation=false, dexing-is-debuggable=true, dexing-min-sdk=23, org.gradle.category=library, org.gradle.jvm.environment=android, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime, org.jetbrains.kotlin.platform.type=androidJvm}.
      > Execution failed for DexingWithClasspathTransform: C:\Users\nikunj\Desktop\work\help_app\build\record_android\intermediates\runtime_library_classes_jar\debug\classes.jar.
         > Error while dexing.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 25s`

Using record 5.2.0

The issue seems to be related to the D8 Dexing process, which is not supporting sealed classes, and there is also a deprecated API warning. Could anyone provide insight into how to resolve this issue?

Thanks in advance!

**Comment by llfbandit** [comment]
https://github.com/llfbandit/record/issues/429#issuecomment-2454703494
